packages/hdwallet-vultisig/src/vultisig.ts

```diff
@@ -0,0 +1,534 @@
+  async getFeatures(): Promise<Record<string, any>> {
+    return {};
+  }
+
+  public async getFirmwareVersion(): Promise<string> {
+    return "vultisig";
+  }
+
+  public async getModel(): Promise<string> {
+    return "Vultisig";
+  }
+
+  public async getLabel(): Promise<string> {
+    return "Vultisig";
+  }
+
+  public async isInitialized(): Promise<boolean> {
+    return true;
+  }
+
+  public async isLocked(): Promise<boolean> {
+    return false;
+  }
+
+  public async clearSession(): Promise<void> {}
+
+  public async initialize(): Promise<void> {}
+
+  public async ping(msg: core.Ping): Promise<core.Pong> {
+    return { msg: msg.msg };
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async sendPin(pin: string): Promise<void> {}
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async sendPassphrase(passphrase: string): Promise<void> {}
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async sendCharacter(charater: string): Promise<void> {}
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async sendWord(word: string): Promise<void> {}
+
+  public async cancel(): Promise<void> {}
+
+  public async wipe(): Promise<void> {}
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async reset(msg: core.ResetDevice): Promise<void> {}
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async recover(msg: core.RecoverDevice): Promise<void> {}
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async loadDevice(msg: core.LoadDevice): Promise<void> {}
+
+  public async disconnect(): Promise<void> {}
+
+  public async getPublicKeys(msg: Array<core.GetPublicKey>): Promise<Array<core.PublicKey | null>> {
+    return await Promise.all(
+      msg.map(async (getPublicKey) => {
+        const { coin, scriptType } = getPublicKey;
+
+        if (scriptType !== undefined) {
+          const isSupported = await this.btcSupportsScriptType(coin, scriptType);
+          if (!isSupported) {
+            return null;
+          }
+        }
+
+        switch (coin) {
+          case "Bitcoin": {
+            // case "Dogecoin": // case "Litecoin": // case "BitcoinCash": // case "Zcash": // case "Dash":
+            // Note this is a pubKey, not an xpub, however vultisig does not support utxo derivation,
+            // so this functions as an account (xpub) for all intents and purposes
+            const pubKey = await this.btcGetAddress({ coin, scriptType } as core.BTCGetAddress);
+            return { xpub: pubKey } as core.PublicKey;
+          }
+          default:
+            break;
+        }
+        throw new Error("Vultisig does not support");
+      })
+    );
+  }
+
+  /** Ethereum */
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public async ethGetAddress(_msg: core.ETHGetAddress): Promise<core.Address | null> {
+    if (this.ethAddress) {
```

Comment:
q: why cache?

This is a copypastaghost job from hdwallet-metamask-multichain, where the author has added super-defensive-but-potentially-useless guard against JSON-RPC spam, since things could get very spammy with MetaMask snaps.

This was *also* added in Phantom, probably as a copypasta job too although not really required there.

Unless we explicitly need cache here, we can keep things clean and remove